### ISSUE TITLE
Styled-Components keyframe animation not triggered V5

### ISSUE THREAD

**Issue by Nosherwan** [styled-components/styled-components#3643]
Hi,
I have posted this on stack overflow before posting it here (https://stackoverflow.com/questions/70359593/styled-components-keyframe-animation-not-triggered-v5), unfortunately havent gotten any resolution for it there.

Below is a link to a running sandbox instance:
(https://codesandbox.io/s/great-frost-ucqre?file=/src/App.js)

I have the following code for a sidebar.
I am trying to simulate a left to right and then from right to left sliding animation on it.
I have tried to google but seems like there is little help for triggering forward and reverse animations using styled-components. It would be great if styled-components documentation includes an example dealing with animation-direction reverse to make it easy for people to follow.

## Steps to reproduce
 I am using styled-components with keyframe animations in nextjs.
The sidebar opens and closes correctly based on the boolean prop passed in, however the keyframes animation is not triggered for either open or close.
## Expected Behavior
the sidebar opening and closing should animate.
## Actual Behavior
The opening and closing animation are not triggered bases on the boolean passed.

```js
import Styled, { css, keyframes } from 'styled-components';

const animate = keyframes`
  0% {
    transform: translateX(-175px);
  }

  100% {
    transform: translateX(0px);
  }
`;

const animated = keyframes`
  100% {
    transform: translateX(0px);
  }
  0% {
    transform: translateX(-175px);
  }
`;
const opened = css`
    animation: ${animate} 0.25s ease-in-out forwards;
`;
const closed = css`
    animation: ${animated} 0.25s ease-in-out;
`;

export const Sidebar = Styled.div<{ show: boolean }>`
    display: flex;
    opacity: ${({ show }) => (show ? 1 : 0)};
    visibility: ${({ show }) => (show ? 'visible' : 'hidden')};
    ${(show) => (show ? opened : closed)};
    animation-direction: ${({ show }) => (show ? 'normal' : 'reverse')};
    top: 0;
    bottom: 0;
    z-index: 1;
    left: 0;
    position: fixed;
    overflow-x: hidden;
    overflow-y: scroll;
    width: 175px;
    flex-direction: column;
    background-color: rgba(239, 239, 239, 0.9);
    align-items: center;
    box-shadow: ${({ show }) =>
        show ? '2px 2px 4px 4px rgba(136, 136, 136, 0.5)' : '0'};
    transition: width 300ms ease;
    color: black;
`;

```

**Comment by webjoyable** [comment]
Having the same issue, is there any explanation why the animation is not getting triggered?

**Comment by gabrielbull** [comment]
You have an error on this line:

```
    ${(show) => (show ? opened : closed)};
```

It should be:

```
    ${({ show }) => (show ? opened : closed)};
```

**Comment by quantizor** [comment]
Closing — this is a support question, stale with no reproduction, or not actionable as a bug report. For help, please use GitHub Discussions or Stack Overflow.
